wurst/_items/Spears.wurst

```diff
@@ -0,0 +1,96 @@
+package Spears
+import ClosureEvents
+import DummyCaster
+import Orders
+import ClosureTimers
+import PublicLibrary
+
+let ABILITY_SPEAR_HIT_DETECTION = 'A0EH'
+
+enum SpearType
+    DarkSpear
+    UltraPoisonSpear
+    RefinedPoisonSpear
+    PoisonSpear
+    SteelSpear
+    IronSpear
+    Spear
+
+let spellToSpearType = new HashMap<int, SpearType>
+let spearListeners = new HashMap<unit, EventListener>()
+
+function onSpear(unit caster, unit target, SpearType spearType)
+    let dummyCaster = new DummyCaster()
+        ..delay(8)
+        ..owner(caster.getOwner())
+        ..origin(caster.getPos())
+    let dummy = dummyCaster.castTarget(ABILITY_SPEAR_HIT_DETECTION, 1, Orders.creepthunderbolt, target)
+
+    bindSpearDamageListener(dummy, target, spearType)
+
+function bindSpearDamageListener(unit dummy, unit target, SpearType spearType)
+    let listener = EventListener.add(EVENT_UNIT_DAMAGED) ->
+        let damaged = GetTriggerUnit()
+        if damaged == target and GetEventDamageSource() == dummy
+            if spearListeners.has(dummy)
+                destroy spearListeners.getAndRemove(dummy)
+                onSpearImpact(target, spearType)
+
+    spearListeners.put(dummy, listener)
+
+function onSpearImpact(unit target, SpearType spearType)
+    switch spearType
+        case UltraPoisonSpear
+            spawnItemChance(0.5, ITEM_ULTRA_POISON_SPEAR, target.getPos())
+        case RefinedPoisonSpear
+            spawnItemChance(0.5, ITEM_REFINED_POISON_SPEAR, target.getPos())
+        case PoisonSpear
+            spawnItemChance(0.5, ITEM_POISON_SPEAR, target.getPos())
+        case DarkSpear
+            spawnItemChance(0.4, ITEM_DARK_SPEAR, target.getPos())
+            ManaBurn(target, GetRandomReal(45, 55))
```

Comment:
Even though I said we shouldn't do a drain over time I still think we should do a resistance similar to `ABILITY_SPEAR_RESIST`

Reply:
Added the system